src/main/kotlin/com/terraformation/backend/tracking/mapbox/MapboxService.kt

```diff
@@ -0,0 +1,90 @@
+package com.terraformation.backend.tracking.mapbox
+
+import com.terraformation.backend.config.TerrawareServerConfig
+import com.terraformation.backend.log.perClassLogger
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.plugins.ClientRequestException
+import io.ktor.client.request.get
+import io.ktor.client.request.post
+import io.ktor.client.request.setBody
+import io.ktor.client.statement.bodyAsText
+import java.net.URL
+import java.time.Instant
+import java.time.InstantSource
+import java.time.temporal.ChronoUnit
+import javax.annotation.ManagedBean
+import kotlinx.coroutines.runBlocking
+
+@ManagedBean
+class MapboxService(
+    private val clock: InstantSource,
+    private val config: TerrawareServerConfig,
+    private val httpClient: HttpClient,
+) {
+  private val log = perClassLogger()
+
+  val enabled: Boolean = config.mapbox.apiToken != null
+
+  /**
+   * The Mapbox username of the account that owns the configured API token. Some Mapbox APIs require
+   * the username to be included explicitly in the request.
+   *
+   * We fetch this lazily the first time it's accessed by requesting information about our access
+   * token, which includes the username.
+   */
+  val username: String by lazy {
+    runBlocking {
+      try {
+        val response: RetrieveTokenResponsePayload = httpClient.get(mapboxUrl("tokens/v2")).body()
+        response.token.user
+      } catch (e: ClientRequestException) {
+        log.error("HTTP ${e.response.status} when fetching Mapbox token information")
+        log.info("Mapbox error response payload: ${e.response.bodyAsText()}")
+        throw MapboxRequestFailedException(e.response.status)
+      }
+    }
+  }
+
+  /** Generates a temporary API token. */
+  fun generateTemporaryToken(): String {
+    if (!enabled) {
+      throw MapboxNotConfiguredException()
+    }
+
+    val url = mapboxUrl("tokens/v2/$username")
+
+    val expirationTime =
+        clock.instant().plus(config.mapbox.temporaryTokenExpirationMinutes, ChronoUnit.MINUTES)
+    val requestPayload =
+        TemporaryTokenRequestPayload(
+            expires = expirationTime, scopes = listOf("styles:tiles", "styles:read", "fonts:read"))
```

Comment:
not sure why this comment didn't come through earlier.
would `styles:tiles` cover the uploaded polygon tileset?

Reply:
Yes, that's my assumption, though I haven't gotten as far as actually creating a tileset yet so it's just a semi-educated guess. It's possible this list of scopes will need to change as we figure things out.